contracts/src/ProtocolAdapter.sol

```diff
@@ -82,18 +81,20 @@ contract ProtocolAdapter is IProtocolAdapter, ReentrancyGuardTransient, Commitme
 
             uint256 nResources = action.logicVerifierInputs.length;
             for (uint256 j = 0; j < nResources; ++j) {
-                Logic.Instance calldata instance = action.logicVerifierInputs[j].instance;
+                Logic.VerifierInput calldata input = action.logicVerifierInputs[j];
 
-                if (instance.isConsumed) {
-                    _addNullifier(instance.tag);
-                } else {
-                    newRoot = _addCommitment(instance.tag);
+                if (input.appData.externalPayload.length != 0) {
```

Comment:
It's a nit but I think this is better split out to a read-only function (`extractForwarderCall` or such), so we only have to write the conditionals once and we can change the representation in the future more easily.

Reply:
Added to task https://github.com/anoma/evm-protocol-adapter/issues/188